### ISSUE TITLE
Flaky-test: test_producer_routing_mode (__main__.PulsarTest)

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#11696]
test_producer_routing_mode (__main__.PulsarTest) is flaky.

[example failure](https://github.com/apache/pulsar/pull/11681/checks?check_run_id=3357855612#step:12:1565)

```
======================================================================
ERROR: test_producer_routing_mode (__main__.PulsarTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "pulsar_test.py", line 648, in test_producer_routing_mode
    message_routing_mode=PartitionsRoutingMode.UseSinglePartition)
  File "/usr/local/lib/python2.7/dist-packages/pulsar/__init__.py", line 603, in create_producer
    p._producer = self._client.create_producer(topic, conf)
Timeout: Pulsar error: TimeOut
```

logs in https://transfer.sh/1xwyvVp/test-logs%20%281%29.zip

the broker logs contain this type of timeout exception (however, this log entry might be unrelated to the flaky test_producer_routing_mode )
```
05:52:49.580 [metadata-store-18-1] WARN  org.apache.pulsar.broker.authorization.AuthorizationService - Producer-client  with Role - anonymous failed to get permissions for topic - persistent://public/default/my-python-test-producer. null
05:52:49.580 [metadata-store-18-1] WARN  org.apache.pulsar.broker.service.Producer - [anonymous] Get unexpected error while autorizing [persistent://public/default/my-python-test-producer]  null
java.util.concurrent.TimeoutException: null
	at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1784) ~[?:1.8.0_292]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1928) ~[?:1.8.0_292]
	at org.apache.pulsar.broker.authorization.AuthorizationService.canProduce(AuthorizationService.java:228) ~[org.apache.pulsar-pulsar-broker-common-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.apache.pulsar.broker.service.Producer.checkPermissions(Producer.java:627) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.apache.pulsar.broker.service.persistent.PersistentTopic.lambda$onPoliciesUpdate$98(PersistentTopic.java:2381) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4707) ~[?:1.8.0_292]
	at org.apache.pulsar.broker.service.persistent.PersistentTopic.onPoliciesUpdate(PersistentTopic.java:2380) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.apache.pulsar.broker.service.BrokerService.lambda$handlePoliciesUpdates$77(BrokerService.java:1785) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_292]
	at org.apache.pulsar.broker.service.BrokerService.lambda$handlePoliciesUpdates$78(BrokerService.java:1785) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:670) ~[?:1.8.0_292]
	at java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:683) ~[?:1.8.0_292]
	at java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2010) ~[?:1.8.0_292]
	at org.apache.pulsar.broker.service.BrokerService.lambda$handlePoliciesUpdates$79(BrokerService.java:1780) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.apache.pulsar.common.util.collections.ConcurrentOpenHashMap$Section.forEach(ConcurrentOpenHashMap.java:387) ~[org.apache.pulsar-pulsar-common-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.apache.pulsar.common.util.collections.ConcurrentOpenHashMap.forEach(ConcurrentOpenHashMap.java:159) ~[org.apache.pulsar-pulsar-common-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.apache.pulsar.broker.service.BrokerService.lambda$handlePoliciesUpdates$80(BrokerService.java:1776) ~[org.apache.pulsar-pulsar-broker-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:670) ~[?:1.8.0_292]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646) ~[?:1.8.0_292]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_292]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975) ~[?:1.8.0_292]
	at org.apache.pulsar.metadata.impl.ZKMetadataStore.lambda$get$6(ZKMetadataStore.java:134) ~[org.apache.pulsar-pulsar-metadata-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty-netty-common-4.1.66.Final.jar:4.1.66.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
05:52:49.581 [metadata-store-18-1] INFO  org.apache.pulsar.broker.service.Producer - [anonymous] is not allowed to produce on topic [persistent://public/default/my-python-test-producer] anymore
05:52:49.581 [metadata-store-18-1] INFO  org.apache.pulsar.broker.service.Producer - Disconnecting producer: Producer{topic=PersistentTopic{topic=persistent://public/default/my-python-test-producer}, client=/127.0.0.1:43282, producerName=standalone-0-422, producerId=0}
05:52:49.582 [metadata-store-18-1] INFO  org.apache.pulsar.broker.service.AbstractTopic - Disabling publish throttling for persistent://public/default/partition-testFlushInPartitionedProducer0_1023509-partition-1

```

**Comment by lhotari** [comment]
similar failure happened again. logs in https://transfer.sh/1uzS8XY/test_logs_test_producer_routing_mode_failure.zip

**Comment by lhotari** [comment]
This might be fixed by #11887 . Closing.
